File: BinarySearch.java
public class BinarySearch {
    public static void main(String[] args)
    {
        int[] arr={10,20,30,40,50,60};
        int n= arr.length;
        int element=40;
        search(arr,0,n-1,element);

    }

    static void search(int[] arr,int first,int last,int element)
    {
        if(first<=last)
        {
            int mid=(first+last)/2;

            if(arr[mid]==element)
            {
                System.out.println("Element found at position "+(mid+1));
            }
            else if(element<arr[mid] )
                search(arr,first,mid-1,element);
            else
                search(arr,mid+1,last,element);
        }
        else
            System.out.println("Element not found.");
            }

}

File: Binarysearch.js
let arr = [1, 2, 3, 4, 5];
let target = 2;

const binarySearch = (arr, target) => {
    let start = 0, end = arr.length - 1;
    while (start <= end) {
        let mid = Math.round((start + end) / 2);
        if (target == arr[mid]) {
            return mid;
        }
        else if (target > arr[mid]) {
            start = mid + 1;
            console.log("hello")
        }
        else {
            end = mid - 1;
        }
    }
    return -1;
}

console.log(binarySearch(arr, target))

